### DISCUSSION TITLE
[Air] - navigateToURL doesn't open pdf on android

### DISCUSSION THREAD

**Discussion by TSerra-PT** [airsdk/Adobe-Runtime-Support#679] [Q&A]
Hi guys,

I have an application developed in flex and I am trying to open a pdf file in android.
In the past this worked, but now (i don't know why) it doesn't worked anymore.
The program/ide doesn't give me any error and the app doesn't open de pdf file.
I use this code:

`try
{
navigateToURL(new URLRequest(File.documentsDirectory.resolvePath("/pdfs/").url+"/test.pdf"));				
}
catch(error:Error)
{
trace(error);
}`

When I'm in debug mode the IDE doesn't hit de catch. It runs normally but doesn't show the pdf.
This only happen in Android, when I try on pc(Windows) works well.
I am using the lastest Air SDK 33.1.1.345. (Maybe this occurs because of the sdk?)

**Comment by marchbold** [comment]
What target sdk are you using?

I'd say that this is because Android doesn't really allow opening file urls anymore.

**Reply by yooni126** [reply]
I'm using AIR 33.1.1.345, and I have the same issue,
 is there any way to open **local** HTML files with `navigateToURL`?

**Reply by ajwfrost** [reply]
Can you confirm your target SDK setting, and perhaps try targeting version 28? We changed to target 29 by default in 33.1.1.217 I believe..
What is the parameter you're passing to `navigateToURL`, and where is the actual file?
thanks

**Reply by yooni126** [reply]
I have written these codes in the manifest file
```
<uses-sdk 
android:minSdkVersion="19"
android:targetSdkVersion="28" />
```
but the problem remains...
and this is my codes :
```
var htmlFile:File = File.documentsDirectory.resolvePath("WebWin//invoice//" + "billing.html");
if(htmlFile.exist)
{
trace("logo is exist")
navigateToURL(new URLRequest(htmlFile.url), "_blank");
}
```
when I try on pc(Windows) works well, but on Android (version 5+) or iOS(version 13.6), it shows me only trace line

**Reply by TSerra-PT** [reply]
I have this code in manifest file:

`			<uses-sdk 
                android:minSdkVersion="15"
                android:targetSdkVersion="29" />`
                
When the file doesn't exists it gives me this error: "requested url is not available"....

**Reply by marchbold** [reply]
Just a suggestion but you might want to look into using an ANE to open the file, using content providers to share the file correctly?

You can use our [Share ANE](https://airnativeextensions.com/extension/com.distriqt.Share) to do this.

**Reply by TSerra-PT** [reply]
> Just a suggestion but you might want to look into using an ANE to open the file, using content providers to share the file correctly?
> 
> You can use our [Share ANE](https://airnativeextensions.com/extension/com.distriqt.Share) to do this.

First of all, thank you for your reply @marchbold 

With this ANE is it possible to open a pdf file, right (ask to the user which pdf reader we wants to open the pdf)?
And this ANE is paid ($15 per year), right ?

**Reply by marchbold** [reply]
Yes, there are two main processes on most mobiles, either passing a file for opening or passing a file for sharing. The processes are very similar but slightly different in their results. The user will be presented with a list of applications that support the filetype and they can select the application. (Or you can use some of the advanced functionality to open a specific application - if installed). 

The initial cost is for a lifetime license, the subscription for subsequent years are discounted 30% and for access to updates and support only (i.e. don't affect your usage of previous versions of the extension).

**Reply by yooni126** [reply]
Thank you for your reply @marchbold 
when I share local HTML files with your share ANE, 
`Share.service.shareFile( htmlFile.url, "html.html", "text/html" );`
on mobile, it doesn't show Safari, or when I select Chrome, it prompts me Chrome cannot handle this link!

also as I said, `navigateToURL(new URLRequest(htmlFile.url), "_blank");` didn't work on mobile devices.

In fact, I want to show the local HTML file in Safari because my users are able to send their HTML files as a PDF in the options menu
![save-safari-page-as-pdf-iphone](https://user-images.githubusercontent.com/3766387/107111584-86a75c00-6866-11eb-92fb-c2b1e14b0802.jpg)

**Reply by marchbold** [reply]
Try sharing the nativePath to the file rather than the url.

**Reply by yooni126** [reply]
I tried but it shows me this prompt
![image0](https://user-images.githubusercontent.com/3766387/107194648-914c2780-6a05-11eb-9a1d-976881798cb2.png)

is there any way to show local HTML files ?

**Reply by marchbold** [reply]
@yooni126 How the apps handle the data you pass them is completely up to the application and beyond our control. My guess would be that chrome has disabled local file:/// urls due to security / privacy issues. 

I believe safari will only actually appear for http(s):// style content. 

I have never tried to share html content like that before, normally I would present it in a webview.
